### ISSUE TITLE
com.hazelcast.jet.core.metrics.MetricsTest.availableViaJmx

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast-jet#1753]
_master_ (commit 880c0cd1582fcb92c6b6cb3cd0ef917de538b2d9)

Failed on IBM JDK 8: http://jenkins.hazelcast.com/job/jet-oss-master-ibm-jdk8/121/testReport/junit/com.hazelcast.jet.core.metrics/MetricsTest/availableViaJmx/

Stacktrace:
```
java.lang.AssertionError: expected:<1000> but was:<827>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.jet.core.metrics.MetricsTest.availableViaJmx(MetricsTest.java:415)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:818)
```

Standard output:
```
Hiccups measured while running test 'availableViaJmx(com.hazelcast.jet.core.metrics.MetricsTest):'
01:59:10, accumulated pauses: 78 ms, max pause: 12 ms, pauses over 1000 ms: 0
01:59:15, accumulated pauses: 103 ms, max pause: 54 ms, pauses over 1000 ms: 0
```

**Comment by cangencer** [comment]
Also failed here, same reason: http://jenkins.hazelcast.com/view/Jet/job/jet-oss-master-zulu-jdk11/131/testReport/junit/com.hazelcast.jet.core.metrics/MetricsTest/availableViaJmx/

**Comment by olukas** [comment]
Originally I thought that this test failure is caused by `assertEquals` used instead of `assertTrueEventually` in `availableViaJmx()`. However even if I changed it to `assertTrueEventually` then it still failed. I checked MBeans manually by `jconsole` during tests and it shows that they really contain incorrect numbers in `total` attribute. Test output shows (because pipeline write to logger) that all expected items were processed by pipeline. It seems like an issue.

**Comment by gurbuzali** [comment]
failed here too http://jenkins.hazelcast.com/job/jet-oss-pr-builder/1434/

**Comment by viliam-durina** [comment]
here too: 
http://jenkins.hazelcast.com/job/jet-oss-pr-builder/1437/
http://jenkins.hazelcast.com/job/jet-oss-pr-builder/1438/

**Comment by blazember** [comment]
Merged the referenced https://github.com/hazelcast/hazelcast/pull/16106 that should fix this failure.

**Comment by olukas** [comment]
Fixed by hazelcast/hazelcast#16106
